fiftyone/core/odm/embedded_document.py

```diff
@@ -40,11 +45,22 @@ class DynamicEmbeddedDocument(
     database.
 
     Dynamic documents can have arbitrary fields added to them.
+
+    Raises FiftyoneDocumentException if a kwarg attribute is reserved.
     """
 
     meta = {"abstract": True, "allow_inheritance": True}
 
     def __init__(self, *args, **kwargs):
+        for key, value in kwargs.items():
+            # Skip MongoDB internal fields
+            if key.startswith('_'):
+                continue
+
+            # Check if the key exists and is a property
```

Comment:
Looks like you don't have [pre-commit hooks](https://github.com/voxel51/fiftyone/blob/develop/CONTRIBUTING.md#pre-commit-hooks) installed

Reply:
Haha oops this is a new conda env I just created :)